test/tests_for_integration/api_to_db_test.py

```diff
@@ -1247,23 +1251,32 @@ def test_run_job_batch(ee2_port, ws_controller, mongo_client):
                     "cell_id": "cid",
                 },
             },
+            "wsid": job_batch_wsid,
             "child_jobs": [],
             "retry_ids": [],
             "retry_saved_toggle": False,
             "batch_job": False,
             "scheduler_id": "123",
             "scheduler_type": "condor",
         }
+        job_wsid = job1.get("wsid")
+        expected_job_wsid = expected_job1.get("wsid")
+
+        job_input_wsid = job1.get("wsid")
+        expected_job_input_wsid = expected_job1.get("wsid")
+
+        assert job_wsid == expected_job_wsid
+        assert job_input_wsid == expected_job_input_wsid
         assert job1 == expected_job1
```

Comment:
This was used to greatly aid in debugging, as it was hard to tell which field didn't match which field from the thrown exception. Do you want to remove it?

Reply:
I get something like this from the test output if there's a difference:
```
E             Full diff:
E               {
E                '_id': ObjectId('60e88ebefa84bddcb48de533'),
E                'authstrat': 'kbaseworkspace',
E                'batch_id': '60e88ebefa84bddcb48de532',
E                'batch_job': False,
E                'child_jobs': [],
E                'job_input': {'method': 'mod.meth',
E                              'narrative_cell_info': {'cell_id': 'cid',
E                                                      'run_id': 'rid',
E                                                      'tag': 'yourit',
E                                                      'token_id': 'tid'},
E                              'params': [{'foo': 'bar'}, 42],
E             -                'requirements': {'clientgroup': 'njs2',
E             ?                                                    -
E             +                'requirements': {'clientgroup': 'njs',
E                                               'cpu': 8,
E                                               'disk': 30,
E                                               'memory': 5},
E                              'service_ver': 'somehash',
E                              'source_ws_objects': ['1/1/1', '1/2/1']},
E                'retry_ids': [],
E                'retry_saved_toggle': False,
E                'scheduler_id': '123',
E                'scheduler_type': 'condor',
E                'status': 'queued',
E                'user': 'nouser',
E               }
```
Do you not get that?

Reply:
No, I see

```
>           assert job1 == expected_job1
E           AssertionError: assert {'_id': ObjectId('60e8ac217ff839f17c75521c'),\n 'authstrat': 'kbaseworkspace',\n 'batch_id': '60e8ac217ff839f17c75521b',\n 'batch_job': False,\n 'child_jobs': [],\n 'job_input': {'method': 'mod.meth',\n               'narrative_cell_info': {'cell_id': 'cid',\n                                       'run_id': 'rid',\n                                       'tag': 'yourit',\n                                       'token_id': 'tid'},\n               'params': [{'foo': 'bar'}, 42],\n               'requirements': {'clientgroup': 'njs',\n                                'cpu': 8,\n                                'disk': 30,\n                                'memory': 5},\n               'service_ver': 'somehash',\n               'source_ws_objects': ['1/1/1', '1/2/1'],\n               'wsid': 2},\n 'retry_ids': [],\n 'retry_saved_toggle': False,\n 'scheduler_id': '123',\n 'scheduler_type': 'condor',\n 'status': 'queued',\n 'user': 'nouser',\n 'wsid': 2} == {'_id': ObjectId('60e8ac217ff839f17c75521c'),\n 'authstrat': 'kbaseworkspace',\n 'batch_id': '60e8ac217ff839f17c75521b',\n 'batch_job': False,\n 'child_jobs': [],\n 'job_input': {'method': 'mod.meth',\n               'narrative_cell_info': {'cell_id': 'cid',\n                                       'run_id': 'rid',\n                                       'tag': 'yourit',\n                                       'token_id': 'tid'},\n               'params': [{'foo': 'bar'}, 42],\n               'requirements': {'clientgroup': 'njs',\n                                'cpu': 8,\n                                'disk': 30,\n                                'memory': 5},\n               'service_ver': 'somehash',\n               'source_ws_objects': ['1/1/1', '1/2/1'],\n               'wsid': 2},\n 'retry_ids': [],\n 'retry_saved_toggle': False,\n 'scheduler_id': '123',\n 'scheduler_type': 'condor',\n 'status': 'queued',\n 'user': 'nouser',\n 'wsid': 'cat'}
```
Unless I run from cmd line, then I see

```
E                'retry_saved_toggle': False,
E                'scheduler_id': '123',
E                'scheduler_type': 'condor',
E                'status': 'queued',
E                'user': 'nouser',
E             -  'wsid': 'cat',
E             ?          ^^^^^
E             +  'wsid': 2,
E             ?          ^
E               }
```
Sometimes there is a diff available and that shows the difference in a GUI viewer

Do you want me to delete it?

Reply:
That's poopy, I guess pycharm doesn't integrate pytest well?

Up to you, but I'd at least note why the lines are there and that they're redundant to the 'master' assert, and I'd also delete or fix one of the two sets of asserts, because they're asserting on the same thing. I think the 2nd one was supposed to assert on the job input wsid but it's not - it's still doing the job wsid.

Reply:
deleted